### ISSUE TITLE
ERROR: insert/update/delete not permitted on chunk

### ISSUE THREAD

**Issue by ismailyenigul** [timescale/timescale-prometheus#87]
I installed timescale-prometheus from timescale-observability helm chart.
There was an issue with prometheus remote_write remote_read configuration. So timescale-prometheus was not receiving the metrics from prometheus for 10 hours.
I updated prometheus settings to point the right timescale-prometheus connector url. It was working fine but after a while I started the receiving the following errors on timescale-prometheus pod.
```
,"level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.635Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1977_685_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":2,"ts":"2020-05-08T13:45:43.652Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1429_394_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.661Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1977_685_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":92,"ts":"2020-05-08T13:45:43.685Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1549_465_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":94,"ts":"2020-05-08T13:45:43.831Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1549_465_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.840Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1549_465_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.857Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1429_394_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":93,"ts":"2020-05-08T13:45:43.893Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1941_669_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.985Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1941_669_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:43.995Z"}
{"caller":"log.go:51","err":"ERROR: insert/update/delete not permitted on chunk \"_hyper_1815_585_chunk\" (SQLSTATE 0A000)","level":"warn","msg":"Error sending samples to remote storage","num_samples":100,"ts":"2020-05-08T13:45:44.061Z"}
```

Errors on timescaledb

```
020-05-08 13:47:04 UTC [15991]: [5eb556b2.3e77-29011] postgres@postgres,app=[unknown] [0A000] HINT:  Make sure the chunk is not compressed.
2020-05-08 13:47:04 UTC [15991]: [5eb556b2.3e77-29012] postgres@postgres,app=[unknown] [0A000] CONTEXT:  COPY apiserver_admission_controller_admission_duration_seconds__1006, line 1
2020-05-08 13:47:04 UTC [15991]: [5eb556b2.3e77-29013] postgres@postgres,app=[unknown] [0A000] STATEMENT:  copy "prom_data"."apiserver_admission_controller_admission_duration_seconds__1006" ( "time", "value", "series_id" ) from stdin binary;
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29014] postgres@postgres,app=[unknown] [0A000] ERROR:  insert/update/delete not permitted on chunk "_hyper_1941_669_chunk"
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29015] postgres@postgres,app=[unknown] [0A000] HINT:  Make sure the chunk is not compressed.
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29016] postgres@postgres,app=[unknown] [0A000] CONTEXT:  COPY kube_statefulset_status_current_revision, line 1
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29017] postgres@postgres,app=[unknown] [0A000] STATEMENT:  copy "prom_data"."kube_statefulset_status_current_revision" ( "time", "value", "series_id" ) from stdin binary;
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29018] postgres@postgres,app=[unknown] [0A000] ERROR:  insert/update/delete not permitted on chunk "_hyper_1429_394_chunk"
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29019] postgres@postgres,app=[unknown] [0A000] HINT:  Make sure the chunk is not compressed.
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29020] postgres@postgres,app=[unknown] [0A000] CONTEXT:  COPY kubelet_runtime_operations_latency_microseconds_count, line 1
2020-05-08 13:47:05 UTC [15991]: [5eb556b2.3e77-29021] postgres@postgres,app=[unknown] [0A000] STATEMENT:  copy "prom_data"."kubelet_runtime_operations_latency_microseconds_count" ( "time", "value", "series_id" ) from stdin binary;
2020-05-08 13:47:05 UTC [16044]: [5eb556d1.3eac-28823] postgres@postgres,app=[unknown] [0A000] ERROR:  insert/update/delete not permitted on chunk "_hyper_1549_465_chunk"
2020-05-08 13:47:05 UTC [16044]: [5eb556d1.3eac-28824] postgres@postgres,app=[unknown] [0A000] HINT:  Make sure the chunk is not compressed.
2020-05-08 13:47:05 UTC [16044]: [5eb556d1.3eac-28825] postgres@postgres,app=[unknown] [0A000] CONTEXT:  COPY kubelet_cgroup_manager_duration_seconds_count, line 1
2020-05-08 13:47:05 UTC [16044]: [5eb556d1.3eac-28826] postgres@postgres,app=[unknown] [0A000] STATEMENT:  copy "prom_data"."kubelet_cgroup_manager_duration_seconds_count" ( "time", "value", "series_id" ) from stdin binary;

```

**Comment by akulkarni** [comment]
Hi @ismailyenigul I notice you have also been active on Slack. Did this get resolved?

**Comment by ismailyenigul** [comment]
Hi @akulkarni 
As far as I know  it is not fixed yet.

**Comment by akulkarni** [comment]
Thanks, taking a look

**Comment by cevian** [comment]
This was fixed in #101

**Comment by ismailyenigul** [comment]
I have the same error with     `Image:          timescale/timescale-prometheus:0.1.0-alpha.4.1`

```
user=postgres database=postgres application_name=pg_isready
2020-06-05 08:05:50 UTC [22319]: [5ed9fcbf.572f-3] postgres@postgres,app=psql [XX000] ERROR:  cannot update/delete rows from chunk "_hyper_1074_546_chunk" as it is compressed
2020-06-05 08:05:50 UTC [22319]: [5ed9fcbf.572f-4] postgres@postgres,app=psql [XX000] CONTEXT:  SQL statement "
	        WITH potentially_drop_series AS (
	            SELECT distinct series_id
	            FROM prom_data.nginx_ingress_controller_request_duration_seconds_bucket
	            WHERE time < '1970-01-01 08:00:00+00'
	            EXCEPT
	            SELECT distinct series_id
	            FROM prom_data.nginx_ingress_controller_request_duration_seconds_bucket
	            WHERE time >= '1970-01-01 08:00:00+00' AND time < '2020-03-07 09:05:19.838393+00'
	        ), confirmed_drop_series AS (
	            SELECT series_id
	            FROM potentially_drop_series
	            WHERE NOT EXISTS (
	                 SELECT 1
	                 FROM  prom_data.nginx_ingress_controller_request_duration_seconds_bucket  data_exists
	                 WHERE data_exists.series_id = potentially_drop_series.series_id AND time >= '2020-03-07 09:05:19.838393+00'
	                 --use chunk append + more likely to find something starting at earliest time
	                 ORDER BY time ASC
	                 LIMIT 1
	            )
	        ), deleted_series AS (
	          DELETE from prom_data_SERIES.nginx_ingress_controller_request_duration_seconds_bucket
	          WHERE id IN (SELECT series_id FROM confirmed_drop_series)
	          RETURNING id, labels
	        )
	        SELECT ARRAY(SELECT DISTINCT unnest(labels) as label_id
	        FROM deleted_series)
	    "
	PL/pgSQL function drop_metric_chunks(text,timestamp with time zone) line 43 at EXECUTE
	SQL statement "SELECT _prom_catalog.drop_metric_chunks(r.metric_name, NOW() - _prom_catalog.get_metric_retention_period(r.metric_name))"
	PL/pgSQL function drop_chunks() line 18 at PERFORM
2020-06-05 08:05:50 UTC [22319]: [5ed9fcbf.572f-5] postgres@postgres,app=psql [XX000] STATEMENT:  CALL prom_api.drop_chunks();

```
